cli/defenseclaw/tui/services/ai_discovery_state.py

```
@@ -449,6 +536,142 @@ def model_derivation(self) -> str:
+        # groups a low-confidence or embedded artifact under the same ID.
+        # The API signal itself must omit model-specific confidence; an
+        # explicit zero remains subject to the normal confidence gate.
+        if self.has_local_model_api_signal_without_discovery_confidence:
+            return True
+
+        reported = self.reported_model_discovery_confidence
+        if reported is not None:
+            if reported < AI_MODEL_RECOMMENDED_MIN_CONFIDENCE:
+                return False
+        elif not self.has_local_model_api_signal:
+            if self.max_signal_confidence < AI_MODEL_RECOMMENDED_MIN_CONFIDENCE:
```

Ruff (0.16.1)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)